### PR TITLE
Replace deprecated OptionMenu APIs with MenuProvider

### PR DESCRIPTION
**Pull request by Varsha-Kulkarni**
## Summary

Replace deprecated OptionMenu APIs with MenuProvider
This closes issue #547 

## Checklist

- [x] Build and linting is passing.
- [x] This change is not breaking existing flow of a system.
- [ ] I have written test case for this change.
- [x] This change is tested from all aspects.

### CODE REVIEW

File: noty-android/app/simpleapp/src/main/java/dev/shreyaspatil/noty/simpleapp/view/detail/NoteDetailFragment.kt [hunk 1]
```diff
@@ -81,9 +85,35 @@ class NoteDetailFragment :
         )
     }
 
-    override fun onCreate(savedInstanceState: Bundle?) {
-        super.onCreate(savedInstanceState)
-        setHasOptionsMenu(true)
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        val menuHost: MenuHost = requireActivity()
```
Comment (PatilShreyas) [inline noty-android/app/simpleapp/src/main/java/dev/shreyaspatil/noty/simpleapp/view/detail/NoteDetailFragment.kt:91]: @Varsha-Kulkarni can we extract different function for this? 
Something like `setupMenus()`

File: noty-android/app/simpleapp/src/main/java/dev/shreyaspatil/noty/simpleapp/view/notes/NotesFragment.kt [hunk 2]
```diff
@@ -51,9 +55,45 @@ class NotesFragment : BaseFragment<NotesFragmentBinding, NotesState, NotesViewMo
 
     private val notesListAdapter by autoCleaned(initializer = { NotesListAdapter(::onNoteClicked) })
 
-    override fun onCreate(savedInstanceState: Bundle?) {
-        super.onCreate(savedInstanceState)
-        setHasOptionsMenu(true)
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        val menuHost: MenuHost = requireActivity()
```
Comment (PatilShreyas) [inline noty-android/app/simpleapp/src/main/java/dev/shreyaspatil/noty/simpleapp/view/notes/NotesFragment.kt:61]: Same here

**Comment by Varsha-Kulkarni** [comment]
Yes!

**Review by PatilShreyas — APPROVED** [review]
Looks good 👍. Thanks @Varsha-Kulkarni

**Pull request merged**